ga4gh/frontend.py

```diff
@@ -372,6 +422,57 @@ def searchVariants(version):
         version, flask.request, app.backend.searchVariants)
 
 
+@app.route('/_login', methods=['GET'])
+def login():
+    session["state"] = rndstr()
+    session["nonce"] = rndstr()
+    args = {
+        "client_id": app.oidcClient.client_id,
+        "response_type": "code",
+        "scope": ["openid", "profile"],
+        "nonce": session["nonce"],
+        "redirect_uri": app.oidcClient.redirect_uris[0],
+        "state": session["state"]
+    }
+    result = app.oidcClient.authorization_request_info(request_args=args)
+    resp = flask.make_response('Redirect', 302)
+    resp.headers['Location'] = result[0]
+    return resp
+
+
+@app.route('/oauth2callback', methods=['GET', 'POST'])
+def oidcCallback():
+    response = dict(request.args.iteritems(multi=True))
+    aresp = app.oidcClient.parse_response(AuthorizationResponse,
+                                          info=response,
+                                          sformat='dict')
+    sessState = session.get('state')
+    respState = aresp['state']
+    if not isinstance(aresp, AuthorizationResponse) or respState != sessState:
```

Comment:
Why would aresp not be an instance of AuthorizationResponse? Since this is security critical, I'd really like to see detailed comments on what can happen here --- paranoia is good!